src/Network/Minio/CopyObject.hs

```diff
@@ -35,7 +35,7 @@ copyObjectInternal b' o srcInfo = do
       sObject = srcObject srcInfo
 
   -- get source object size with a head request
-  (ObjectInfo _ _ _ srcSize) <- headObject sBucket sObject
+  (ObjectInfo _ _ _ srcSize _) <- headObject sBucket sObject
```

Comment:
Use the `oiSize` record accessor to fetch the size, instead of pattern matching here, as `ObjectInfo` could change in the future and break this pattern match.